operators/src/mock/mock_data_set_data_source.rs

```diff
@@ -0,0 +1,175 @@
+use geoengine_datatypes::primitives::{Coordinate2D, TimeInterval};
+use geoengine_datatypes::spatial_reference::SpatialReferenceOption;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+// TODO: generify this to support all data types
+#[derive(Debug, Clone)]
+pub struct MockDataSetDataSourceLoadingInfo {
+    pub points: Vec<Coordinate2D>,
+}
+
+impl LoadingInfo<MockDataSetDataSourceLoadingInfo, VectorResultDescriptor>
+    for MockDataSetDataSourceLoadingInfo
+{
+    fn get(&self, _query: QueryRectangle) -> Result<MockDataSetDataSourceLoadingInfo> {
+        Ok(self.clone()) // TODO: intersect points with query rectangle
+    }
+
+    fn meta(&self) -> Result<VectorResultDescriptor> {
+        Ok(VectorResultDescriptor {
+            data_type: VectorDataType::MultiPoint,
+            spatial_reference: SpatialReferenceOption::Unreferenced,
+        })
+    }
+
+    fn box_clone(
+        &self,
+    ) -> Box<dyn LoadingInfo<MockDataSetDataSourceLoadingInfo, VectorResultDescriptor>> {
+        Box::new(self.clone())
+    }
+}
+
+// impl LoadingInfoProvider<MockDataSetDataSourceLoadingInfo, VectorResultDescriptor>
+//     for MockExecutionContext
+// {
+//     fn loading_info(
+//         &self,
+//         _data_set: &DataSetId,
+//     ) -> Result<Box<dyn LoadingInfo<MockDataSetDataSourceLoadingInfo, VectorResultDescriptor>>>
+//     {
+//         Ok(Box::new(self.loading_info.as_ref().unwrap().clone())
+//             as Box<
+//                 dyn LoadingInfo<MockDataSetDataSourceLoadingInfo, VectorResultDescriptor>,
+//             >)
+//     }
+// }
+
+pub struct MockDataSetDataSourceProcessor {
+    loading_info: Box<dyn LoadingInfo<MockDataSetDataSourceLoadingInfo, VectorResultDescriptor>>,
+}
+
+impl QueryProcessor for MockDataSetDataSourceProcessor {
+    type Output = MultiPointCollection;
+    fn query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        _ctx: &'a dyn QueryContext,
+    ) -> BoxStream<'a, Result<MultiPointCollection>> {
+        // TODO: split into `chunk_byte_size`d chunks
+        // let chunk_size = ctx.chunk_byte_size() / std::mem::size_of::<Coordinate2D>();
+
+        let loading_info = self.loading_info.get(query).unwrap();
+
+        stream::once(async move {
+            Ok(MultiPointCollection::from_data(
+                loading_info.points.iter().map(Into::into).collect(),
+                vec![TimeInterval::default(); loading_info.points.len()],
+                HashMap::new(),
+            )?)
+        })
+        .boxed()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct MockDataSetDataSourceParams {
+    pub data_set: DataSetId,
+}
+
+pub type MockDataSetDataSource = SourceOperator<MockDataSetDataSourceParams>;
+
+#[typetag::serde]
+impl VectorOperator for MockDataSetDataSource {
+    fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<InitializedVectorOperator>> {
```

Comment:
dyn?

Reply:
please elaborate

Reply:
i thought InitializedVectorOperator is a trait so it should be a Box<dyn InitializedVectorOperator>

Reply:
it is a trait but it is a type alias:
`pub type InitializedVectorOperator =
    dyn InitializedOperator<VectorResultDescriptor, TypedVectorQueryProcessor>;`

so the dyn is hidden here. If I try to add it I get
> Expected trait, found type alias `InitializedVectorOperator`

Reply:
ah well then we keep it like this